### PR TITLE
add a new module to configure multicast global.

### PR DESCRIPTION
**Pull request by yuandongx**
##### SUMMARY

add a new module to configure multicast global.

##### ISSUE TYPE

- New Module Pull Request

### test case
```
  - name: check vrf about vrf not exist
    ce_multicast_global:
      aftype: v4
      vrf: test
      state: present

```
### test log
```

changed: [10.190.121.125] => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
    "end_state": {
        "multicast_global": [
            {
                "addressFamily": "ipv4unicast",
                "vrfName": "test"
            }
        ]
    },
    "existing": {
        "multicast_global": []
    },
    "invocation": {
        "module_args": {
            "aftype": "v4",
            "state": "present",
            "vrf": "test"
        }
    },
    "proposed": {
        "addressFamily": "ipv4unicast",
        "state": "present",
        "vrfName": "test"
    },
    "updates": [
        "multicast routing-enable"
    ]
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************
10.190.121.125             : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
### show on real device
```
[*8850_130.12-vpn-instance-test]di th
#
ip vpn-instance test
 ipv4-family
  route-distinguisher 12:12
  multicast routing-enable
 ipv6-family
  route-distinguisher 13:13
#
return
[*8850_130.12-vpn-instance-test]


```
### test case
```

  - name: absent vrf about vrf not exist
    ce_multicast_global:
      aftype: v4
      vrf: test
      state: absent

```
### test log
```

changed: [10.190.121.125] => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": true,
    "end_state": {
        "multicast_global": []
    },
    "existing": {
        "multicast_global": [
            {
                "addressFamily": "ipv4unicast",
                "vrfName": "test"
            }
        ]
    },
    "invocation": {
        "module_args": {
            "aftype": "v4",
            "state": "absent",
            "vrf": "test"
        }
    },
    "proposed": {
        "addressFamily": "ipv4unicast",
        "state": "absent",
        "vrfName": "test"
    },
    "updates": [
        "undo multicast routing-enable"
    ]
}
META: ran handlers
META: ran handlers

PLAY RECAP ************************************************************************************************
10.190.121.125             : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
### show on real device
```
[*8850_130.12-vpn-instance-test]di th
#
ip vpn-instance test
 ipv4-family
  route-distinguisher 12:12
 ipv6-family
  route-distinguisher 13:13
#
return
[*8850_130.12-vpn-instance-test]
```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    def set_change_state(self):
+        """set change state"""
+        state = self.state
+        change = False
+        self.get_multicast_global()
+        # new or edit
+        if state == 'present':
+            if not self.multicast_global_info['multicast_global']:
+                # self.multicast_global_info['multicast_global'] has not value
```
Comment (yuandongxx) [inline lib/ansible/modules/network/cloudengine/ce_multicast_global.py]: unused code?
```suggestion
                # multicast_global_info has no multicast_global value.
```

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                change = True
+        else:
+            # delete
+            if self.multicast_global_info['multicast_global']:
+                # self.multicast_global_info['multicast_global'] has value
```
Comment (yuandongxx) [inline lib/ansible/modules/network/cloudengine/ce_multicast_global.py]:
```suggestion
                # multicast_global_info has no multicast_global value
```

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        xml_str = xml_str.replace('\r', '').replace('\n', ''). \
+            replace('xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"', ""). \
+            replace('xmlns="http://www.huawei.com/netconf/vrp"', "")
+        root = ElementTree.fromstring(xml_str)
+        mcast_enable = root.findall(
+            "mcastbase/mcastAfsEnables/mcastAfsEnable")
+        if mcast_enable:
+            # mcast_enable = [{vrfName:11,addressFamily:'xx'},{vrfName:22,addressFamily:'xx'}...]
+            for mcast_enable_key in mcast_enable:
+                # mcast_enable_key = {vrfName:11,addressFamily:'xx'}
```
Comment (yuandongxx) [inline lib/ansible/modules/network/cloudengine/ce_multicast_global.py]:
```suggestion
                # i.e. mcast_enable_key = {vrfName:11,addressFamily:'xx'}
```

**Review by yuandongxx — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py (hunk 3 above)
Comment (yuandongx) [reply lib/ansible/modules/network/cloudengine/ce_multicast_global.py]: Done

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py (hunk 2 above)
Comment (yuandongx) [reply lib/ansible/modules/network/cloudengine/ce_multicast_global.py]: Done

File: lib/ansible/modules/network/cloudengine/ce_multicast_global.py (hunk 1 above)
Comment (yuandongx) [reply lib/ansible/modules/network/cloudengine/ce_multicast_global.py]: Done

**Comment by xuxiaowei0512** [comment]
ready_for_review

**Comment by hwDCN** [comment]
shipit

**Review by yuandongxx — APPROVED** [review]

**Comment by hwDCN** [comment]
shipit

**Pull request merged**